src/request-data/user-preferences/normalizers.js

```diff
@@ -33,6 +33,21 @@ export class SitePreferenceNormalizer extends PreferenceNormalizer {
   static projectSortMode(val) {
     return ['alphabetical', 'latest', 'newest'].includes(val) ? val : 'latest';
   }
+
+  static outdatedVersionWarningDismissDate(val) {
+    // Frontend to Backend
+    if (val instanceof Date) {
+      return val.toISOString();
+    }
+
+    // Backend to Frontend
+    const isoDateRegex = /^\d{4}-\d{2}-\d{2}([T](\d{2}:\d{2}:\d{2}(\.\d+)?Z)?)?$/;
```

Comment:
I feel like there's one more `?` here than I'd expect. Isn't this saying that you could have `T` on its own without numbers after it?

Reply:
I have removed all `?` as [Data.toISOString()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString) returns a simplified string which is always 24 or 27 character. In our case it will be always 24.